apps/src/templates/MultipleSectionsAssigner.jsx

```diff
@@ -0,0 +1,324 @@
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+import {connect} from 'react-redux';
+import color from '@cdo/apps/util/color';
+import i18n from '@cdo/locale';
+import BaseDialog from '@cdo/apps/templates/BaseDialog';
+import Button from '@cdo/apps/templates/Button';
+import {sectionForDropdownShape} from '@cdo/apps/templates/teacherDashboard/shapes';
+import TeacherSectionOption from './TeacherSectionOption';
+import {
+  assignToSection,
+  unassignSection
+} from '@cdo/apps/templates/teacherDashboard/teacherSectionsRedux';
+import {updateHiddenScript} from '@cdo/apps/code-studio/hiddenLessonRedux';
+
+class MultipleSectionsAssigner extends Component {
+  static propTypes = {
+    courseId: PropTypes.number,
+    assignmentName: PropTypes.string.isRequired,
+    onClose: PropTypes.func.isRequired,
+    forceReload: PropTypes.bool,
+    courseOfferingId: PropTypes.number,
+    courseVersionId: PropTypes.number,
+    scriptId: PropTypes.number,
+    reassignConfirm: PropTypes.func,
+    isOnCoursePage: PropTypes.bool,
+    isStandAloneUnit: PropTypes.bool,
+    participantAudience: PropTypes.string,
+    // Redux
+    sections: PropTypes.arrayOf(sectionForDropdownShape).isRequired,
+    unassignSection: PropTypes.func.isRequired,
+    assignToSection: PropTypes.func.isRequired,
+    updateHiddenScript: PropTypes.func.isRequired,
+    selectedSectionId: PropTypes.number
+  };
+
+  constructor(props) {
+    super(props);
+
+    let initialSectionsAssignedToCourseList = [];
+
+    // check to see if this is coming from the UNIT landing page - if so add courses featuring this unit
+    if (!this.props.isOnCoursePage) {
+      if (this.props.isStandAloneUnit) {
+        for (let i = 0; i < this.props.sections.length; i++) {
+          if (
+            this.props.courseVersionId ===
+            this.props.sections[i].courseVersionId
+          ) {
+            initialSectionsAssignedToCourseList.push(this.props.sections[i]);
+          }
+        }
+      } else {
+        for (let i = 0; i < this.props.sections.length; i++) {
+          if (this.props.scriptId === this.props.sections[i].unitId) {
+            initialSectionsAssignedToCourseList.push(this.props.sections[i]);
+          }
+        }
+      }
+    } else if (this.props.isOnCoursePage) {
+      // checks to see if this is coming from the COURSE landing page
+      for (let i = 0; i < this.props.sections.length; i++) {
+        if (this.props.courseId === this.props.sections[i].courseId) {
```

Comment:
You can probably check `courseVersionId` here as well. If it works with `courseId`, it's probably not worth changing though.